addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -212,15 +235,406 @@ class EnvironmentMountService extends Service {
     ]);
   }
 
+  /**
+   * Function that calculates which users need should and should not have access permissions for the given studyId
+   * Accordingly calls methods that ensure/remove/update permissions for said users
+   *
+   * @param {String} studyId
+   * @param {Object} updateRequest - permission add/remove/update requests coming from SWB UI
+   */
+  async manageWorkspacePermissions(studyId, updateRequest) {
```

Comment:
I have a few design comments here, specially regarding the handling of exceptions and how you want to async  this.   It might be better to discuss this over chime.

Reply:
You might also need to pass the requestContext because the applyWorkspacePermissions() method might be used somewhere else and we need to check that permissions are enforced there too.

Reply:
Sounds good, thanks.

Reply:
Exception handling has been added. User permissions are being checked on studyPermissionService.verifyRequestorAccess() as we discovered :)